💡 Doc/models

ARYAN-NIKNEZHAD opened the pull request:
This pull request addresses the documentation issues by adding detailed docstrings to the models.

Changes:

Added comprehensive docstrings to all models for better clarity and understanding.

Closes #6

ARYAN-NIKNEZHAD committed:
:sparkles: feat(mixins): Add-BaseLocation-abstract-model

ARYAN-NIKNEZHAD committed:
:bulb: docs(models): Add comprehensive docstrings to location models
- Added detailed docstrings to the following models: Province, County, District, City, RuralDistrict, and Village.
- Ensured that all models have clear, informative docstrings explaining their purpose and attributes.

These changes improve the documentation within the codebase, making it easier for developers to understand the purpose and structure of each model.

ARYAN-NIKNEZHAD committed:
:bug: fix(code-review): fix the code review changes
1. Add  and
2. fix the related_name in ReralDistrict Model

ARYAN-NIKNEZHAD committed:
:bulb: docs(models): Add comprehensive docstrings to location models Closes #6

sepehr-akbarzadeh commented on iranian_cities/mixins/base_location.py:
```diff
@@ -0,0 +1,31 @@
+    """
+    Abstract base model for location entities.
+
+    Attributes:
+        name (str): The name of the location.
+        code (int): The unique code for the location.
```
On model docstr we dont need attribute explanation. Because we explain each field with two powerful attributes of django. So, remove attribute explanation.

Inside docstr of models we just explain role of entity inside app context. Clarify concept of the model. Just this not more.

ARYAN-NIKNEZHAD replied:
@sepehr-akbarzadeh, thank you for your valuable feedback! 🎉

I've implemented the changes based on your suggestions. The docstrings for all models have been updated to focus solely on their roles within the app context, and detailed attribute explanations have been removed to utilize Django's field documentation.

Please review the updated code and let me know if there are any further adjustments needed. 🙏

ARYAN-NIKNEZHAD committed:
:ok_hand: fix(models.py): Refactor model docstrings for clarity and conciseness

- Updated docstrings for all models to focus on their role within the app context.
- Removed detailed attribute explanations to leverage Django's built-in field documentation.
- Ensured each model's purpose and context are clearly explained.

This enhances readability and maintains concise documentation, making the codebase cleaner and easier to understand.
